UnhollowerBaseLib/MemoryUtils.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using UnhollowerRuntimeLib.XrefScans;
+
+namespace UnhollowerBaseLib
+{
+    internal class MemoryUtils
+    {
+        public struct SignatureDefinition
+        {
+            public string pattern;
+            public string mask;
+            public int offset;
+            public bool xref;
+        }
+        public static unsafe void* FindSignatureInModule(ProcessModule module, SignatureDefinition sigDef)
```

Comment:
Is there a specific need to use `void*`? If you need to reference to a memory address but still retain the easy pointer arithmetic, maybe you can use `nint` or `nuint` types instead? They have direct implicit cast to IntPtr and pointer types but have the benefit of being managed types and still having the direct pointer arithmetic. 

With `nint` you can generally get rid of a bunch of needless casts to `void*`.

Reply:
Thank you, I was not aware that I could use `nint`/`nuint` instead. It definitely looks a lot cleaner than having a bunch of pointer casts